pkg/login/social/generic_oauth.go

```diff
@@ -79,13 +81,13 @@ func (s *SocialGenericOAuth) IsOrganizationMember(client *http.Client) bool {
 	return false
 }
 
-// searchJSONForEmail searches the provided JSON response for an e-mail address
+// searchJSONForAttr searches the provided JSON response for the given attribute
 // using the configured e-mail attribute path associated with the generic OAuth
 // provider.
 // Returns an empty string if an e-mail address is not found.
-func (s *SocialGenericOAuth) searchJSONForEmail(data []byte) string {
-	if s.emailAttributePath == "" {
-		s.log.Error("No e-mail attribute path specified")
+func (s *SocialGenericOAuth) searchJSONForAttr(attribute string, data []byte) string {
```

Comment:
Can still name the argument `attributePath` I think to make it more descriptive.

Reply:
Meant that you could rename attribute argument:
```suggestion
func (s *SocialGenericOAuth) searchJSONForAttr(attributePath string, data []byte) string {
```